google-cloud-storage/src/main/java/com/google/cloud/storage/UnbufferedWritableByteChannelSession.java

```diff
@@ -28,12 +28,26 @@ interface UnbufferedWritableByteChannelSession<ResultT>
   interface UnbufferedWritableByteChannel extends WritableByteChannel, GatheringByteChannel {
     @Override
     default int write(ByteBuffer src) throws IOException {
-      return Math.toIntExact(write(new ByteBuffer[] {src}));
+      return Math.toIntExact(write(new ByteBuffer[] {src}, 0, 1));
```

Comment:
Does `0, 1` represent there's only 1 buffer right now?

Reply:
Correct. These values represent the index to start from in the provided array and how many indices to consume.

This is reducing a stack entry where before it would do `write(ByteBuffer) -> write(ByteBuffer[]) -> write(ByteBuffer[], int, int)`. While logically equivalent (and the JVM can probably inline the method to reduce impact anyway) doing this makes sure we don't add another stack frame.